File: Easy/3442. Maximum Difference Between Even and Odd Frequency I/Python/main.py
from collections import Counter

class Solution:
    def maxDifference(self, s: str) -> int:
        freq = Counter(s)

        old_frequencies = []
        even_frequencies = []

        for count in freq.values():
            if count % 2 == 0:
                even_frequencies.append(count)
            else:
                old_frequencies.append(count)

        
    max_odd = max(old_frequencies)
    min_even = min(even_frequencies)

    return max_odd - min_even

File: Easy/3442. Maximum Difference Between Even and Odd Frequency I/Rust/main.rs
use std::collections::HashMap;

impl Solution {
    pub fn max_difference(s: String) -> i32 {
        let mut freq = HashMap::new();

        for ch in s.chars() {
            *freq.entry(ch).or_insert(0) += 1;
        }

        let mut odd_freqs = Vec::new();
        let mut even_freqs = Vec::new();

        for &count in freq.values() {
            if count % 2 == 0 {
                even_freqs.push(count);
            } else {
                odd_freqs.push(count);
            }
        }

        let max_odd = *odd_freqs.iter().max().unwrap();
        let min_even = *even_freqs.iter().min().unwrap();

        max_odd - min_even
    }

